docs/pipeline-configuration-files/specifications.mdx

```diff
@@ -0,0 +1,272 @@
+          - id: proc1           # Processor ID [required]
+            type: js            # Processor type [required]
+            workers: 2          # Number of parallel workers
+            settings:           # A map of configuration keys and values for the processor (specific to the chosen type)
+              script: >
+                function process(record) {
+                  return record; // pass through
+                }
+    processors:                 # A list of processor configurations, processors are attached to the pipeline
+      - id: proc2               # Processor ID [required]
+        type: hoistfieldpayload # Processor type [required]
+        settings:               # A map of configuration keys and values for the processor (specific to the chosen type)
+          field: ${ENV_VAR}     # You can use environment variables by wrapping them in a dollar sign and curly braces ${}
+    dead-letter-queue:          # Dead-letter queue (DLQ) configuration
+      plugin: "builtin:file"    # DLQ Connector plugin
+      settings:                 # A map of configuration keys and values for the plugin (specific to the chosen plugin)
+        path: "./dlq.out"
+      window-size: 5            # DLQ nack window size
+      window-nack-threshold: 2  # DLQ nack window threshold
+```
+
+## version
+
+- **Data Type**: String
+- **Required**: No
+- **Default**: latest (`2.0`)
+- **Allowed Values**: `1.0`, `1.1`, `2.0`
+- **Description**: Version defines the schema version for the pipeline configuration file and controls which parser is
+  used to decode it.
+
+## pipelines
+
+- **Data Type**: Sequence node containing [`pipeline`](#pipeline) nodes
+- **Required**: Yes
+- **Default**: None
+- **Description**: The node contains a list of [`pipeline`](#pipeline) definitions. Pipelines will be provisioned in the
+  order in which they are defined.
+
+## pipeline
+
+- **Data Type**: Mapping node
+- **Required**: n/a
+- **Default**: None
+- **Description**: This node contains the configuration of a single pipeline.
+
+### id
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: ID of the [`pipeline`](#pipeline). It should uniquely identify the pipeline across all pipelines. If
+  multiple pipelines use the same ID, none of them will be provisioned.
+
+**Warning**: Changing this property will cause all connectors to start from the beginning.
+
+### status
+
+- **Data Type**: String
+- **Required**: No
+- **Default**: `stopped`
+- **Allowed Values**: `stopped`, `running`
+- **Description**: This field controls the status of the pipeline at startup. If set to `running` the pipeline will be
+  automatically started, if set to `stopped` the pipeline will be provisioned without being started.
+
+### name
+
+- **Data Type**: String
+- **Required**: No
+- **Default**: Same as pipeline [`id`](#id)
+- **Description**: Human readable name for the pipeline. Needs to be unique across all pipelines.
```

Comment:
Can you please remind me why it has to be unique (when we already have the ID), especially because connector names for example do not need to be unique? I vaguely recall something about our platform and fetching pipelines.

Reply:
IIRC it's because we are using the pipeline name in metrics to distinguish them, since names are provided by the user and IDs were not at the time we designed this (they still are not if the pipeline is created through the API). So to prevent mixing metrics between multiple pipelines they need to have unique names.